.github/workflows/node.js.yml

```diff
@@ -0,0 +1,28 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
```

Comment:
We should likely pin these Actions to specific SHAs with a comment next to them for the version that matches. Dependabot should help keep those updated based on [this blog](https://github.blog/changelog/2022-10-31-dependabot-now-updates-comments-in-github-actions-workflows-referencing-action-versions/).

Reply:
I'd also review the [security hardening guide for GitHub Actions](https://docs.github.com/en/enterprise-cloud@latest/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions) to ensure we align to best practices.

We may want to validate these changes using [OSSF Scorecard](https://github.com/ossf/scorecard). It might be good to eventually add OSSF Scorecard as a GitHub Action to this repo.

Reply:
Added SHAs to versions.

Think we should skip scorecard for now until OSPO has a better stance there.